src/actionCreators.js

```diff
@@ -31,9 +31,33 @@ export const fetchRootResourceTemplate = resourceTemplateId => (dispatch) => {
   dispatch(retrieveResourceTemplateStarted(resourceTemplateId))
 
   getResourceTemplate(resourceTemplateId, 'ld4p').then((response) => {
-    dispatch(rootResourceTemplateLoaded(response.response.body))
+    // If resource template loads, then validate.
+    const resourceTemplate = response.response.body
+    const reason = validateResourceTemplate(resourceTemplate)
+    if (reason) {
+      dispatch(retrieveError(resourceTemplateId, reason))
+    } else {
+      dispatch(rootResourceTemplateLoaded(resourceTemplate))
+    }
   }).catch((err) => {
     console.error(err)
     dispatch(retrieveError(resourceTemplateId))
   })
 }
+
+export const validateResourceTemplate = (resourceTemplate) => {
```

Comment:
Can you move this to a different file, because it is not an actionCreator (it does not return a function that takes dispatch and optionally getState)

Reply:
Done.